code/datums/components/rotation.dm

```diff
@@ -83,87 +70,106 @@
 	return COMPONENT_NOTRANSFER
 
 /datum/component/simple_rotation/UnregisterFromParent()
-	remove_verbs()
-	remove_signals()
+	RemoveVerbs()
+	RemoveSignals()
 	. = ..()
 
 /datum/component/simple_rotation/Destroy()
-	QDEL_NULL(can_user_rotate)
-	QDEL_NULL(can_be_rotated)
-	QDEL_NULL(after_rotation)
+	QDEL_NULL(AfterRotation)
 	//Signals + verbs removed via UnRegister
 	. = ..()
 
 /datum/component/simple_rotation/ClearFromParent()
-	remove_verbs()
+	RemoveVerbs()
 	return ..()
 
 /datum/component/simple_rotation/proc/ExamineMessage(datum/source, mob/user, list/examine_list)
 	SIGNAL_HANDLER
+	examine_list += span_notice("Alt + Right-click to rotate it clockwise. Alt + Left-click to rotate it counterclockwise.")
+	if(rotation_flags & ROTATION_REQUIRE_WRENCH)
+		examine_list += span_notice("This requires a wrench to be rotated.")
 
-	if(rotation_flags & ROTATION_ALTCLICK)
-		examine_list += span_notice("Alt-click to rotate it clockwise.")
-
-/datum/component/simple_rotation/proc/HandRot(datum/source, mob/user, rotation = default_rotation_direction)
+/datum/component/simple_rotation/proc/RotateRight(datum/source, mob/user)
 	SIGNAL_HANDLER
+	Rotate(user, ROTATION_CLOCKWISE)
 
-	if(!can_be_rotated.Invoke(user, rotation) || !can_user_rotate.Invoke(user, rotation))
-		return
-	BaseRot(user, rotation)
-
-/datum/component/simple_rotation/proc/WrenchRot(datum/source, obj/item/I, mob/living/user)
+/datum/component/simple_rotation/proc/RotateLeft(datum/source, mob/user)
 	SIGNAL_HANDLER
-
-	if(!can_be_rotated.Invoke(user,default_rotation_direction) || !can_user_rotate.Invoke(user,default_rotation_direction))
+	Rotate(user, ROTATION_COUNTERCLOCKWISE)
+
+/datum/component/simple_rotation/proc/Rotate(mob/user, degrees)
+	if(QDELETED(user))
+		stack_trace("[src] is being rotated [user ? "with a qdeleting" : "without a"] user")
+	if(!istype(user))
```

Comment:
```suggestion
	else if(!istype(user))
```
You didn't answer on the other comment.
If we are not using `CRASH()` then we need to avoid a duplicate runtime.
I still think these two should be using `CRASH()` instead of `stack_trace()`, no?
In case you don't know, `CRASH()` generates a runtime and returns. `stack_trace()` generates a runtime and carries on without returning.

Reply:
I was not familiar with the debugging behavior with DM code.  We need to use `CRASH()` for this then.